src/org/omegat/core/data/RealProject.java

```diff
@@ -714,6 +714,12 @@ public void compileProjectAndCommit(String sourcePattern, boolean doPostProcessi
         // them to
         // the repository (BUGS#1176)
         CoreEvents.fireProjectChange(IProjectEventListener.PROJECT_CHANGE_TYPE.COMPILE);
+        while (CoreEvents.isInProjectEventLoop) {
```

Comment:
Could you please implement a generic method that fire event and wait completion, and call the new method instead of calling `CoreEvent.fireProjectChange` function?